engine/framework/registry/registry.go

```diff
@@ -43,6 +44,9 @@ type Registry interface {
 		config []byte,
 		epoch frameModel.Epoch,
 	) (framework.Worker, error)
+	// AnalyzeError parses error, checks whether this error is retryable and
+	// returns rpc error based on it.
+	AnalyzeError(err error, tp framework.WorkerType) error
```

Comment:
I slightly prefer we don't add this function to Registry. `AnalyzeError` is related to a specific WorkerType, and inside `CreateWorker` we resolved the WorkerType to WorkerFactory, we can call WorkerFactory.IsRetryableError at that place.

Reply:
addressed. I have separated the logic of `AnalyzeError` and leave `IsRetryableError` API in registry only.